packages/applications-service-api/src/utils/document.mapper.js

```diff
@@ -0,0 +1,131 @@
+const config = require('../lib/config');
+const toCamelCase = require('lodash.camelcase');
+
+const LABEL_MAPPING = {
+	stage: {
+		// NI mapping
+		1: 'Pre-application',
+		2: 'Acceptance',
+		3: 'Pre-examination',
+		4: 'Examination',
+		5: 'Recommendation',
+		6: 'Recommendation and Decision',
+		7: 'Post-decision',
+
+		// back office mapping
+		'pre-application': 'Pre-application',
+		acceptance: 'Acceptance',
+		'pre-examination': 'Pre-examination',
+		examination: 'Examination',
+		recommendation: 'Recommendation',
+		decision: 'Decision',
+		post_decision: 'Post-decision',
+		developers_application: "Developer's Application"
+	},
+	category: {
+		developersapplication: "Developer's Application"
+	}
+};
+
+const mapDocumentFilterLabel = (filterName, filterValue) => {
+	try {
+		if (filterName === 'category') {
+			// normalise filter name as NI DB contains mess of different values
+			const normalisedKey = filterValue.toLowerCase().replace(/[^a-zA-Z0-9]/g, '');
+			return LABEL_MAPPING[filterName][normalisedKey] || filterValue;
+		}
+
+		return LABEL_MAPPING[filterName][filterValue];
+	} catch (e) {
+		return filterValue;
+	}
+};
+
+const mapDocuments = (documents) => {
+	const attributesToLowerCase = (document) =>
+		Object.keys(document).reduce((memo, key) => {
+			let value = document[key];
+
+			if (key === 'path') value = config.documentsHost.concat(value);
+
+			memo[toCamelCase(key)] = value;
+
+			return memo;
+		}, {});
+
+	return documents.map(attributesToLowerCase);
+};
+
+const mapBackOfficeDocuments = (documents) =>
+	documents.map((document) => ({
+		id: document.id,
+		dataId: document.documentReference,
+		caseReference: document.caseRef,
+		stage: document.stage,
+		type: document.documentType,
+		filter1: document.filter1,
+		filter2: document.filter2,
+		description: document.description,
+		size: document.size,
+		mime: document.mime,
+		path: document.documentURI,
+		// status: 'Published', // should always be Published, so not needed in API response?
+		datePublished: document.datePublished,
+		// deadlineDate: "", // no equivalent in BO schema
+		// personalName: '', // no equivalent in BO schema
```

Comment:
I think personalName is the one field I'd like us to understand a bit more.  The FE uses this at the moment mainly as a description type field.  Not much we can do as it isn't passed by BO so it is hopefully covered by the other fields (description).

Reply:
From looking at the test data (would be good to see prod data for comparison), `personalName` looks to be much like `author`, which seems to be null on all the test data I've seen.

We have `author` in BO, so would suggest switching the frontend to use that instead of `personalName`.